python/paddle/fluid/initializer.py

```diff
@@ -500,15 +514,24 @@ def __call__(self, var, block):
             std = np.sqrt(2.0 / float(fan_in + fan_out))
             op = block._prepend_op(
                 type="gaussian_random",
-                outputs={"Out": var},
+                outputs={"Out": out_var},
                 attrs={
-                    "shape": var.shape,
-                    "dtype": int(var.dtype),
+                    "shape": out_var.shape,
+                    "dtype": out_dtype,
                     "mean": 0.0,
                     "std": std,
                     "seed": self._seed
                 },
                 stop_gradient=True)
+
+        if var.dtype == VarDesc.VarType.FP16:
+            block.append_op(
+                type="cast",
+                inputs={"X": out_var},
+                outputs={"Out": var},
+                attrs={"in_dtype": out_var.dtype,
+                       "out_dtype": var.dtype})
```

Comment:
Wether the other initializer needs to do this FP32 weight creating and casting?

Reply:
Yes. They all need. And all initializers have been made to support float16.